sign compare issues in usnic provider

When configuring with "./configure CFLAGS=-Wsign-compare", there are a bunch of warnings in the usnic provider.  Filing this here because it may be a week or three before we can address these, and we didn't want to forget.

```sh
  CC       prov/usnic/src/usdf_av.lo
prov/usnic/src/usdf_av.c: In function 'usdf_am_insert_async':
prov/usnic/src/usdf_av.c:274:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  for (i = 0; i < count; i++) {
                ^
prov/usnic/src/usdf_av.c: In function 'usdf_am_insert_sync':
prov/usnic/src/usdf_av.c:346:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  for (i = 0; i < count; i++) {
                ^
  CC       prov/usnic/src/usdf_cm.lo
prov/usnic/src/usdf_cm.c: In function 'usdf_cm_msg_connect_cb_rd':
prov/usnic/src/usdf_cm.c:305:11: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if (ret != entry_len) {
           ^
  CC       prov/usnic/src/usdf_cq.lo
  CC       prov/usnic/src/usdf_dgram.lo
prov/usnic/src/usdf_dgram.c: In function 'usdf_dgram_recvv':
prov/usnic/src/usdf_dgram.c:119:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  for (i = 0; i < count; ++i) {
                ^
prov/usnic/src/usdf_dgram.c: In function 'usdf_dgram_sendv':
prov/usnic/src/usdf_dgram.c:231:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  for (i = 0; i < count; i++) {
                ^
prov/usnic/src/usdf_dgram.c: In function 'usdf_dgram_prefix_recvv':
prov/usnic/src/usdf_dgram.c:409:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  for (i = 0; i < count; ++i) {
                ^
  CC       prov/usnic/src/usdf_domain.lo
prov/usnic/src/usdf_domain.c: In function 'usdf_dom_rdc_free_data':
prov/usnic/src/usdf_domain.c:103:45: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   while (atomic_get(&udp->dom_rdc_free_cnt) < 
                                             ^
  CC       prov/usnic/src/usdf_endpoint.lo
  CC       prov/usnic/src/usdf_ep_dgram.lo
  CC       prov/usnic/src/usdf_ep_msg.lo
prov/usnic/src/usdf_ep_msg.c: In function 'usdf_tx_msg_enable':
prov/usnic/src/usdf_ep_msg.c:115:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  for (i = 0; i < tx->tx_attr.size; ++i) {
                ^
prov/usnic/src/usdf_ep_msg.c: In function 'usdf_rx_msg_enable':
prov/usnic/src/usdf_ep_msg.c:183:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  for (i = 0; i < qp->uq_rq.urq_num_entries - 1; ++i) {
                ^
prov/usnic/src/usdf_ep_msg.c:199:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  for (i = 0; i < rx->rx_attr.size; ++i) {
                ^
  CC       prov/usnic/src/usdf_ep_rdm.lo
prov/usnic/src/usdf_ep_rdm.c: In function 'usdf_tx_rdm_enable':
prov/usnic/src/usdf_ep_rdm.c:116:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  for (i = 0; i < tx->tx_attr.size; ++i) {
                ^
prov/usnic/src/usdf_ep_rdm.c: In function 'usdf_rx_rdm_enable':
prov/usnic/src/usdf_ep_rdm.c:184:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  for (i = 0; i < qp->uq_rq.urq_num_entries - 1; ++i) {
                ^
prov/usnic/src/usdf_ep_rdm.c:200:16: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  for (i = 0; i < rx->rx_attr.size; ++i) {
                ^
  CC       prov/usnic/src/usdf_eq.lo
  CC       prov/usnic/src/usdf_fabric.lo
  CC       prov/usnic/src/usdf_mem.lo
  CC       prov/usnic/src/usdf_msg.lo
  CC       prov/usnic/src/usdf_pep.lo
prov/usnic/src/usdf_pep.c: In function 'usdf_pep_read_connreq':
prov/usnic/src/usdf_pep.c:245:11: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if (ret != entry_len) {
           ^
```

Heads up to @goodell @xuywang

--- a/prov/usnic/src/usdf_av.c
+++ b/prov/usnic/src/usdf_av.c
@@ -231,7 +231,7 @@ usdf_am_insert_async(struct fid_av *fav, const void *addr, size_t count,
 	struct usdf_av *av;
 	struct usdf_fabric *fp;
 	int ret;
-	int i;
+	size_t i;
 
 	USDF_TRACE_SYS(AV, "\n");
 
@@ -337,7 +337,7 @@ usdf_am_insert_sync(struct fid_av *fav, const void *addr, size_t count,
 	struct usdf_dest *dest;
 	int ret_count;
 	int ret;
-	int i;
+	size_t i;
 
 	USDF_TRACE_SYS(AV, "\n");
 

--- a/prov/usnic/src/usdf_cm.c
+++ b/prov/usnic/src/usdf_cm.c
@@ -310,7 +310,7 @@ usdf_cm_msg_connect_cb_rd(void *v)
 		ret = usdf_eq_write_internal(ep->ep_eq, FI_CONNECTED, entry,
 				entry_len, 0);
 		free(entry);
-		if (ret != entry_len) {
+		if (ret != (int)entry_len) {
 			free(ep->e.msg.ep_dest);
 			ep->e.msg.ep_dest = NULL;
 			usdf_cm_msg_connreq_failed(crp, ret);

--- a/prov/usnic/src/usdf_dgram.c
+++ b/prov/usnic/src/usdf_dgram.c
@@ -102,7 +102,7 @@ usdf_dgram_recvv(struct fid_ep *fep, const struct iovec *iov, void **desc,
 	struct usd_recv_desc rxd;
 	struct usd_qp_impl *qp;
 	uint32_t index;
-	int i;
+	size_t i;
 
 	ep = ep_ftou(fep);
 	qp = to_qpi(ep->e.dg.ep_qp);
@@ -223,7 +223,7 @@ usdf_dgram_sendv(struct fid_ep *fep, const struct iovec *iov, void **desc,
 	uint8_t *copybuf;
 	size_t len;
 	struct iovec send_iov[USDF_DGRAM_MAX_SGE];
-	int i;
+	size_t i;
 
 	ep = ep_ftou(fep);
 	dest = (struct usd_dest *)(uintptr_t) dest_addr;
@@ -396,7 +396,7 @@ usdf_dgram_prefix_recvv(struct fid_ep *fep, const struct iovec *iov,
 	struct usd_recv_desc rxd;
 	struct usd_qp_impl *qp;
 	uint32_t index;
-	int i;
+	size_t i;
 
 	ep = ep_ftou(fep);
 	qp = to_qpi(ep->e.dg.ep_qp);

--- a/prov/usnic/src/usdf_domain.c
+++ b/prov/usnic/src/usdf_domain.c
@@ -103,7 +103,7 @@ usdf_dom_rdc_free_data(struct usdf_domain *udp)
 
 		/* XXX probably want a timeout here... */
 		while (atomic_get(&udp->dom_rdc_free_cnt) < 
-		       udp->dom_rdc_total) {
+		       (int)udp->dom_rdc_total) {
 			pthread_yield();
 		}
 

--- a/prov/usnic/src/usdf_ep_msg.c
+++ b/prov/usnic/src/usdf_ep_msg.c
@@ -76,7 +76,7 @@ usdf_tx_msg_enable(struct usdf_tx *tx)
 	struct usdf_cq_hard *hcq;
 	struct usd_filter filt;
 	int ret;
-	int i;
+	size_t i;
 
 	udp = tx->tx_domain;
 
@@ -164,7 +164,7 @@ usdf_rx_msg_enable(struct usdf_rx *rx)
 	uint8_t *ptr;
 	size_t mtu;
 	int ret;
-	int i;
+	size_t i;
 
 	udp = rx->rx_domain;
 

--- a/prov/usnic/src/usdf_ep_rdm.c
+++ b/prov/usnic/src/usdf_ep_rdm.c
@@ -76,7 +76,7 @@ usdf_tx_rdm_enable(struct usdf_tx *tx)
 	struct usdf_cq_hard *hcq;
 	struct usd_filter filt;
 	int ret;
-	int i;
+	size_t i;
 
 	USDF_TRACE_SYS(EP_CTRL, "\n");
 
@@ -163,7 +163,7 @@ usdf_rx_rdm_enable(struct usdf_rx *rx)
 	uint8_t *ptr;
 	size_t mtu;
 	int ret;
-	int i;
+	size_t i;
 
 	USDF_TRACE_SYS(EP_CTRL, "\n");
 

--- a/prov/usnic/src/usdf_msg.c
+++ b/prov/usnic/src/usdf_msg.c
@@ -316,7 +316,7 @@ ssize_t
 usdf_msg_sendv(struct fid_ep *fep, const struct iovec *iov, void **desc,
                  size_t count, fi_addr_t dest_addr, void *context)
 {
-	int i;
+	size_t i;
 	struct usdf_ep *ep;
 	struct usdf_tx *tx;
 	struct usdf_msg_qe *wqe;
@@ -370,7 +370,7 @@ usdf_msg_sendv(struct fid_ep *fep, const struct iovec *iov, void **desc,
 ssize_t
 usdf_msg_sendmsg(struct fid_ep *fep, const struct fi_msg *msg, uint64_t flags)
 {
-	int i;
+	size_t i;
 	struct usdf_ep *ep;
 	struct usdf_tx *tx;
 	struct usdf_msg_qe *wqe;
@@ -511,7 +511,7 @@ usdf_msg_inject(struct fid_ep *fep, const void *buf, size_t len,
 ssize_t
 usdf_msg_recvmsg(struct fid_ep *fep, const struct fi_msg *msg, uint64_t flags)
 {
-	int i;
+	size_t i;
 	struct usdf_ep *ep;
 	struct usdf_rx *rx;
 	struct usdf_msg_qe *rqe;

--- a/prov/usnic/src/usdf_pep.c
+++ b/prov/usnic/src/usdf_pep.c
@@ -251,7 +251,7 @@ usdf_pep_read_connreq(void *v)
 		ret = usdf_eq_write_internal(pep->pep_eq, FI_CONNREQ, entry,
 				entry_len, 0);
 		free(entry);
-		if (ret != entry_len) {
+		if (ret != (int)entry_len) {
 			usdf_cm_msg_connreq_failed(crp, ret);
 			return 0;
 		}

--- a/prov/usnic/src/usdf_rdm.c
+++ b/prov/usnic/src/usdf_rdm.c
@@ -538,7 +538,7 @@ ssize_t
 usdf_rdm_sendv(struct fid_ep *fep, const struct iovec *iov, void **desc,
                  size_t count, fi_addr_t dest_addr, void *context)
 {
-	int i;
+	size_t i;
 	struct usdf_ep *ep;
 	struct usdf_tx *tx;
 	struct usdf_rdm_qe *wqe;
@@ -607,7 +607,7 @@ usdf_rdm_sendv(struct fid_ep *fep, const struct iovec *iov, void **desc,
 ssize_t
 usdf_rdm_sendmsg(struct fid_ep *fep, const struct fi_msg *msg, uint64_t flags)
 {
-	int i;
+	size_t i;
 	struct usdf_ep *ep;
 	struct usdf_tx *tx;
 	struct usdf_rdm_qe *wqe;

